test/test_sort_key.rb

```diff
@@ -0,0 +1,20 @@
+# frozen_string_literal: true
+
+require "test/unit"
+require "bcdice"
+require "bcdice/game_system"
+
+class TestSortKey < Test::Unit::TestCase
+  HIRAGANA_BASIC_CHARS = "[あいうえおかきくけこさしすせそたちつてとなにぬねのはひふへほまみむめもやゆよらりるれろわをん]"
+
+  data do
+    BCDice.all_game_systems.map { |s| [s::ID, s::SORT_KEY] }.to_h
+  end
+  def test_sort_key(sort_key)
+    assert_match(
+      /^#{HIRAGANA_BASIC_CHARS}+[\d.A-Za-z]*|国際化:[\w ]+:.+|\*たいすほつと$/,
```

Comment:
この正規表現ではすり抜けるパターンがありました。例えば `SORT_KEY = "AAAそおと\nわあると"` が通ります。`|` の対象範囲を `(?: )` で囲むことと、行頭・行末を `\A \z` で示すことが必要です。

```suggestion
      /\A(?:#{HIRAGANA_BASIC_CHARS}+[\d.A-Za-z]*|国際化:[\w ]+:.+|\*たいすほつと)\z/,
```